src/mudlet.cpp

```diff
@@ -194,6 +193,11 @@ mudlet::mudlet()
 
     scanForMudletTranslations(QStringLiteral(":/lang"));
     scanForQtTranslations(getMudletPath(qtTranslationsPath));
+
+    if (firstLaunch()) {
```

Comment:
BTW `readEarlySettings(...)` will default `mInterfaceLanguage` to `en_US` if there is no stored setting - which there won't be on first launch. So you might as well put the `autodetectPreferredLanguage()` call in as the second argument to the call to `QSettings::value(...)` there - and you don't even need the `firstLaunch()` check there - just change:
```CPP
        mInterfaceLanguage = settings.value("interfaceLanguage", QStringLiteral("en_US")).toString();
```
to:
```CPP
        mInterfaceLanguage = settings.value("interfaceLanguage", autodetectPreferredLanguage()).toString();
```

:nerd_face:

Reply:
That makes testing even easier as well - just delete the `interfaceLanguage` line from the `Mudlet.conf` file in the `~/.config/mudlet` directory!

Reply:
That would complicate things because it'll turn into "to test feature A, just change this. To test feature B, just change that. To test feature C, just change that, and that, and that" whereas the current first launch instructions are simple and straightforward: https://wiki.mudlet.org/w/Manual:Testing#Simulate_a_fresh_install

But the two aren't mutually exclusive, so I'll update this code, and the instructions can stay the same, win-win.

Reply:
Updated.